test/integration/kayrock/record_batch_test.exs

```diff
@@ -234,7 +287,7 @@ defmodule KafkaEx.KayrockRecordBatchTest do
 
     fetch_responses =
       KafkaEx.fetch(topic, 0,
-        offset: max(offset - 2, 0),
+        offset: offset,
```

Comment:
I don't remember exactly why this was set this way - probably some kind of bizarre test behavior.  Did you have a compelling reason to change it?

Reply:
At first I tried to understand the reason to get a previous offset instead the one returned by the producer, but since it always worked before I kept is AS-IS. After I have introduced the headers the kayrock tests start failing randomly but always with the same reason `nil` message.

I watch the topic with Kafka tool and them I figure out that the offset - 2 doesn't match with the expected message then I tried to remove the `max(offset - 2, 0)` and then I got success on every execution what make things that is the reason and any other test failed

I don't have a rational about that behavior only that on my mind the offset should not be manipulated to get the desired message, it was just a recursively try-n-check.

Reply:
@dantswain in case you're interested, here is the commit where you added this `max(offset - 2, 0)`:  7197a909922ff5a2f75de4ee445ea9b9099395ac

I didn't see anything in particular to help me understand. IMO the new changed code makes more sense :shrug: